yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -236,10 +238,39 @@ export class AvmPersistableStateManager {
   }
 
   /**
-   * Get a contract's bytecode from the contracts DB
+   * Get a contract's bytecode from the contracts DB, also trace the contract class and instance
    */
   public async getBytecode(contractAddress: AztecAddress, selector: FunctionSelector): Promise<Buffer | undefined> {
-    return await this.worldStateDB.getBytecode(contractAddress, selector);
+    let exists = true;
+    // If the bytecode is not found, we let the executor decide that to do
+    const bytecode = await this.worldStateDB.getBytecode(contractAddress, selector);
+    let contractInstance = await this.worldStateDB.getContractInstance(contractAddress);
+    // If the contract instance is not found, we assume it has not be deployed. We will also be unable to find the
+    // contract class as we will not have the id. While the class might exist, we hopefully won't need it to generate a proof (tbd).
+    if (contractInstance === undefined) {
+      exists = false;
+      contractInstance = SerializableContractInstance.empty().withAddress(contractAddress);
+      this.trace.traceExecutionStart(
+        bytecode ?? Buffer.alloc(1),
```

Comment:
So I'm not really sure what the behavior should be if we can't find the bytecode - do we throw (because we can't execute the instructions in the simulator anyways)?

If not we should hint some representation of "empty". I think the circuit can't put nothing in the column because then it wouldn't be provable.

Reply:
I think we decided that if you call a contract that doesn't exist, the call instruction should just do nothing and return success=0 with empty returndata.

So do whatever's easiest for the bytecode hashing, because then we'll just throw out the results and prove that the contract address passed to the call instruction doesn't exist in the nullifier tree.